File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_num = 0
        current = 0
        for num in nums:
            if num == 1: 
                current += 1
            else:
                if current > max_num:
                    max_num = current
                current = 0
        return max_num if max_num > current else current

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0, curr = 0;
        for (int num : nums) {
            if (num == 1) {
                curr++;
                if (curr > ans) ans = curr;
            } else {
                curr = 0;
            }
        } 
        return ans;
    }
};